docs/bundle_format.rst

```diff
@@ -0,0 +1,106 @@
+to allow interoperability between environments:
+
+* There are no constraints on how the files in a Bundle can be organized.
+* OLX files must use the **.olx** extension. Files with a different extension may be ignored or not treated as
+  OLX by plugins and clients.
+* A bundle.json file must be added by authoring clients which specifies the shareable components, bundle
+  dependencies and assets. See below for details.
+* Every XBlock element in an OLX file must have an **id** attribute. The value of this attribute will be used
+  to construct the **block_key**. In a single Bundle, all ids must be unique. Blockstore will not validate ids however.
+* Nested XBlocks in separate OLX files can be referenced by adding a **path** attribute to the element which points
+  to the relevant OLX file. Blockstore does not parse or validate any paths itself. However plugins may do so.
+* XBlock Runtimes should provide a service which can be used to read a file given a **path** or to get a
+  public URL for it (if it is specified in assets field). XBlocks will be responsible for replacing paths
+  in any assets and resources they are using with URLs.
+
+Paths
+-------------------------
+
+Paths are URIs of the form **[//bundle_version_alias]path**.
+
+bundle.json
+-------------------------
+The bundle.json file contains metadata about the Bundle. Authoring clients must add and update this themselves.
+It contains the following fields:
+
+meta
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+  "meta": {
+    "version": 1
+  },
+
+
+main_component
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **main_component** field can be used to specify the file path to an XBlock which is the primary XBlock
+of the Bundle that authors should use.
+
+.. code-block::
+
+    "main_component": "/unit1.olx"
+
+components
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **components** field can be used to specify file paths for XBlocks in the Bundle the author wants
+available for use independently. They may be shown in previews of the Bundle so that authors can select them but
+will have lower priority than the XBlock specified in **main_component**.
+
+.. code-block::
+
+    "components": {
+      "/video_teaser.olx",
+      "/mcqs/mcq1.olx",
+      "/mcqs/mcq2.olx",
+      "/mcqs/mcq3.olx"
+    }
+
+It is valid to not specify **main_component** and **components**. This may be useful in some cases like having a
+shared Bundle with branding assets only.
+
+dependencies
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **dependencies** field is a dictionary of **bundle_version_alias: bundle_version** pairs. If a file from
+another Bundle is to be referenced, a version of that Bundle must be specified in this field.
+
+.. code-block::
+
+  "dependencies": {
+    "a": 21d45e735e134c41ae3b24fde26d4369@8,
+    "b": b97c9907ecd54f4eb5f4c7eb51dd58e3@2
+  }
+
+The XBlock Runtime will provide a service which allows reading files from Bundles.
```

Comment:
I would actually prefer just to have a new field type like `StaticAssetField` which holds an opaque string, and which provides methods to the XBlock like `get_absolute_url()`, `get_source_path_string()` (returns `/static/foo.blah` for assets stored in contentstore as today's platform does, or `//blah/foo` for Blockstore assets).

By not having a dedicated field API for this today, we run into several issues, and I think a `StaticAssetField` would solve them.

Just as we don't want Blockstore to know about XBlocks, I don't think we want XBlocks to have to know about Blockstore, and especially not to use a Blockstore service.

Reply:
Ah, good idea! But I think we will need something on the Runtime as well which has the config, right?

Reply:
Yes, the runtime would have to know how to get the absolute URL for a given blockstore static asset path.